common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -98,6 +106,26 @@ public SharedPreferencesAccountCredentialCache(
         mCacheValueDelegate = accountCacheValueDelegate;
     }
 
+    /**
+     * Returns the list of all AccountRecords.
+     */
+    private List<AccountRecord> getAllAccountRecords() {
+        if (mAccountList.isEmpty()) {
+            fetchAndPopulateAllAccountsAndCredentials();
+        }
+        return Collections.unmodifiableList(mAccountList);
+    }
+
+    /**
+     * Returns the list of all Credentials.
+     */
+    private List<Credential> getAllCredentials() {
+        if (mCredentialList.isEmpty()) {
+            fetchAndPopulateAllAccountsAndCredentials();
+        }
+        return Collections.unmodifiableList(mCredentialList);
+    }
+
     @Override
     public synchronized void saveAccount(@NonNull final AccountRecord accountToSave) {
```

Comment:
>saveAccount [](http://example.com/codeflow?start=29&length=11)

After saveAccount & saveCredential also the in-memory cache lists should be updated. otherwise, getAccounts won't return all values.